src/simple.ts

```diff
@@ -9,34 +9,33 @@ export interface SimpleLocatorParameters {
   defaultPort?: number;
 }
 
-export function simpleLocator(parameters: string): Locator.FacetLocator;
-export function simpleLocator(parameters: SimpleLocatorParameters): Locator.FacetLocator;
-export function simpleLocator(parameters: any): Locator.FacetLocator {
-  if (typeof parameters === "string") parameters = { resource: parameters };
-  var resource: string = parameters.resource;
-  var defaultPort: number = parameters.defaultPort;
-  if (!resource) throw new Error("must have resource");
-
-  var locations = resource.split(";").map((locationString) => {
-    var parts = locationString.split(":");
-    if (parts.length > 2) throw new Error("invalid resource part '" + locationString + "'");
-
-    var location: Locator.Location = {
-      host: parts[0]
-    };
-    if (parts.length === 2) {
-      if (!integerRegExp.test(parts[1])) {
-        throw new Error("invalid port in resource '" + parts[1] + "'");
+export function simpleLocatorFactory(params: SimpleLocatorParameters) {
+  function simpleLocator(parameters: any): Locator.FacetLocator {
+    var resource: string = parameters.resource;
+    var defaultPort: number = parameters.defaultPort;
+    if (!resource) throw new Error("must have resource");
+
+    var locations = resource.split(";").map((locationString) => {
+      var parts = locationString.split(":");
+      if (parts.length > 2) throw new Error("invalid resource part '" + locationString + "'");
+
+      var location: Locator.Location = {
+        host: parts[0]
+      };
+      if (parts.length === 2) {
+        if (!integerRegExp.test(parts[1])) {
+          throw new Error("invalid port in resource '" + parts[1] + "'");
+        }
+        location.port = Number(parts[1]);
+      } else if (defaultPort) {
+        location.port = defaultPort;
       }
-      location.port = Number(parts[1]);
-    } else if (defaultPort) {
-      location.port = defaultPort;
-    }
 
-    return location;
-  });
+      return location;
+    });
 
-  return (callback) => {
-    callback(null, locations[Math.floor(Math.random() * locations.length)]);
-  };
+    return (callback) => {
+      callback(null, locations[Math.floor(Math.random() * locations.length)]);
+    };
+  } return simpleLocator(params);
```

Comment:
This is wrong. `params` supplied to `simpleLocatorFactory` is different from what `simpleLocator` requires. You should just do `return simpleLocator;`

Also, can you put this return statement on the new line?